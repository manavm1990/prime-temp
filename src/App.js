import React from 'react';
import Input from "./components/Input";
import { tryConvert } from './utils';

export default function App() {
  const [tempC, setTempC] = React.useState(0);
  const [tempF, setTempF] = React.useState(0);

  const handleChange = (e) => {
    const { value, id } = e.target;

    if (id === "temp-c") {
      setTempC(value);
      setTempF(tryConvert(value, "toFahrenheit"));
    } else{
      setTempF(value);
      setTempC(tryConvert(value, "toCelsius"));
    }
  };

  return (
    <main>
      <Input id="temp-f" label="Temperature (°F)" changeHandler={handleChange} value={tempF}/>
      <Input id="temp-c" label="Temperature (°C)" changeHandler={handleChange} value={tempC}/>
    </main>
  );
}
