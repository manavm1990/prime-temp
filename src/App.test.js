import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

it("renders without crashing", () => {
  const appTree = renderer.create(<App />);
  expect(appTree).toMatchSnapshot();
})

it("converts to 째C to 째F", () => {
  render(<App />);
  const cInput = screen.getByLabelText("Temperature (째C)");
  userEvent.type(cInput, "100");

  const fInput = screen.getByLabelText("Temperature (째F)");
  expect(fInput.value).toBe("212");
})