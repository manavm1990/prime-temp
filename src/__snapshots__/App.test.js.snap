// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<main>
  <div>
    <label
      htmlFor="temp-f"
    >
      Temperature (째F)
    </label>
    <input
      id="temp-f"
      onChange={[Function]}
      placeholder="Temperature (째F)"
      type="number"
      value={0}
    />
  </div>
  <div>
    <label
      htmlFor="temp-c"
    >
      Temperature (째C)
    </label>
    <input
      id="temp-c"
      onChange={[Function]}
      placeholder="Temperature (째C)"
      type="number"
      value={0}
    />
  </div>
</main>
`;
